File: src/main/java/com/hatmani/videostreamingsys/Dto/MovieDto.java
package com.hatmani.videostreamingsys.Dto;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@AllArgsConstructor
@NoArgsConstructor
@Data
public class MovieDto {
    private String id;
    private String moviename;
    private String description;
    private String fileurl;
    private String imageurl;
   // @JsonIgnore()
    private String categorieid;
    private String duration;
    private String releaseyear;


}

File: MovieWeb/moviestreaming/src/app/models/MovieDTO.ts
export class MovieDTO { 

    id?: any;
    moviename?: string;
    description?: string;
    categorieid?: string;
    duration?: string;
    releaseyear?:string;
    fileurl?:string;
    imageurl?:File;
    
    

    
  }
  export class MovieDTORequest { 
   
    moviename?: string;
    description?: string;
    categorieid?: string;
    
    duration?: string;
    releaseyear?:string;
    

    
  }
  export class categoryDTO{
    id?:string;
    name?:string;
    description?: string;
movies?:MovieDTO[]|null=null;
  }
  export class Pagecategory{
  
    content?:categoryDTO[];
    pageNumber?:number;
    pageSize?:number;
    totalElements?:number;
    totalPages?:number;
    first?:boolean;
    last?:boolean;
  }
  export class PageMovie{
    content?:MovieDTO[];
    pageNumber?:number;
    pageSize?:number;
    totalElements?:number;
    totalPages?:number;
    first?:boolean;
    last?:boolean;
  }